pkg/geoipupdate/database/http_reader.go

```diff
@@ -176,6 +176,7 @@ func (reader *HTTPDatabaseReader) download(
 			return "", time.Time{}, false, fmt.Errorf("unexpected HTTP status code: %w", err)
 		}
```

Comment:
Is there any reason to keep the `if err == nil` branch now? It looks like we could combine them and ignore the ReadAll error.